discovery-handlers/debug-echo/src/discovery_handler.rs

```diff
@@ -23,12 +27,23 @@ pub const DEBUG_ECHO_AVAILABILITY_CHECK_PATH: &str = "/tmp/debug-echo-availabili
 /// String to write into DEBUG_ECHO_AVAILABILITY_CHECK_PATH to make Other devices undiscoverable
 pub const OFFLINE: &str = "OFFLINE";
 
+
+#[derive(Serialize, Debug)]
+pub struct QueryDevicePostBody {
+    pub id: String,
+    pub protocol: String
+}
+
 /// DebugEchoDiscoveryDetails describes the necessary information needed to discover and filter debug echo devices.
 /// Specifically, it contains a list (`descriptions`) of fake devices to be discovered.
 /// This information is expected to be serialized in the discovery details map sent during Discover requests.
 #[derive(Serialize, Deserialize, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct DebugEchoDiscoveryDetails {
+    //if there is query_device_http in the discovery detail, discovery handler will call agent QueryDeviceInfo rpc function for each discovered new device
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub query_device_http: Option<String>,    
```

Comment:
if every discovery handle is going to contain `query_device_http`, can this be moved under `DiscoveryHandler` rather than as part of the `discovery_details` which are unique to each discovery handler?

Reply:
I definitely agree. As this is the first PR I work on Akri, I tried to not change too much, for example, not changing the yaml structure and HELM. I plan this as a feature implementation with minimum change to other part.